template/src/ontology/Makefile.jinja2

```diff
@@ -550,8 +550,9 @@ component-download-{{ component.filename }}: | $(TMPDIR)
 	remove {% if component.base_iris is not none %}{% for iri in component.base_iris %}--base-iri {{iri}} {% endfor %}{% else %}--base-iri $(OBOBASE)/{{ project.id.upper() }} {% endif %}--axioms external --preserve-structure false --trim false \{% endif %}
 	annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $(TMPDIR)/$@.owl; fi
 
-$(COMPONENTSDIR)/{{ component.filename }}:  component-download-{{ component.filename }}
-	if [ $(COMP) = true ] ; then if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $@ ; then echo "Component identical."; else echo "Component is different, updating." && cp $(TMPDIR)/component-download-{{ component.filename }}.owl $@; fi; fi
+$(COMPONENTSDIR)/{{ component.filename }}: component-download-{{ component.filename }}
+	if [ $(COMP) = true ]; then if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $(TMPDIR)/component-download-{{ component.filename }}.tmp.owl ; then echo "Component identical."; \
```

Comment:
What would happen on a “clean” repo if there is no `$(TMPDIR)/component-download- {{ component.filename }}.tmp.owl` file from a previous run?

Reply:
Good point.. Maybe `if not exist touch`?

Reply:
Either that or something like “if the file does not exist, then go to the ‘not-identical’ branch”.

Reply:
I tested in an uberon branch without suggested changes. I removed all files `tmp/component-download-*` and it worked as expected, "go to the 'not-identical' branch".

```bash
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://raw.githubusercontent.com/hubmapconsortium/ccf-validation-tools/master/owl/UB_ASCTB_subset.owl \
annotate --ontology-iri http://purl.obolibrary.org/obo/uberon/component-download-hra_subset.owl annotate -V http://purl.obolibrary.org/obo/uberon/releases/2024-01-03/component-download-hra_subset.owl --annotation owl:versionInfo 2024-01-03 -o tmp/component-download-hra_subset.owl.owl; fi
if [ true  = true ]; then if cmp -s tmp/component-download-hra_subset.owl.owl tmp/component-download-hra_subset.owl.tmp.owl ; then echo "Component identical."; \
      else echo "Component is different, updating." && cp tmp/component-download-hra_subset.owl.owl tmp/component-download-hra_subset.owl.tmp.owl && robot --catalog catalog-v001.xml annotate -i tmp/component-download-hra_subset.owl.owl --ontology-iri http://purl.obolibrary.org/obo/uberon/components/hra_subset.owl annotate -V http://purl.obolibrary.org/obo/uberon/releases/2024-01-03/components/hra_subset.owl --annotation owl:versionInfo 2024-01-03 -o components/hra_subset.owl; fi; fi
Component is different, updating.
```

Reply:
Yep, you are right @anitacaron:

```
➜  ontology git:(fix-abbrev) ✗ cmp -s gorgon.owl hp-edit.owl || echo "FAIL"
FAIL
➜  ontology git:(fix-abbrev) ✗ cmp -s hp-edit.owl hp-edit.owl || echo "FAIL" 
➜  ontology git:(fix-abbrev) ✗ 
```

Seems when the file does not exist, cmp evaluates to false, which is the desired behaviour.

I think we can close this comment.